cli/azd/internal/vsrpc/environment_service_create.go

```diff
@@ -0,0 +1,121 @@
+package vsrpc
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"io/fs"
+	"os"
+	"path/filepath"
+
+	"github.com/azure/azure-dev/cli/azd/internal/appdetect"
+	"github.com/azure/azure-dev/cli/azd/pkg/apphost"
+	"github.com/azure/azure-dev/cli/azd/pkg/environment"
+	"github.com/azure/azure-dev/cli/azd/pkg/environment/azdcontext"
+	"github.com/azure/azure-dev/cli/azd/pkg/tools/dotnet"
+)
+
+// CreateEnvironmentAsync is the server implementation of:
+// ValueTask<bool> CreateEnvironmentAsync(Session, Environment, IObserver<ProgressMessage>, CancellationToken);
+func (s *environmentService) CreateEnvironmentAsync(
+	ctx context.Context, sessionId Session, newEnv *Environment, observer IObserver[ProgressMessage],
```

Comment:
Would `newEnv Environment` work just as well? I just wonder if `*Environment` could create the possibility for the client to omit it in the request, resulting in a nil `panic`. An empty struct may mean instead that the error "Name cannot be empty" is returned.